File: online-judge-solutions/Codeforces/408A.cpp
//https://codeforces.com/problemset/problem/408/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	200 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int n,x;
	while(cin>>n){
		vector<int> k(n);
		for(int i=0;i<n;i++){cin>>k[i];}
		vector<vector<int>> m;
		for(int i=0;i<k.size();i++){
			vector<int> v;
			for(int j=0;j<k[i];j++){
				cin>>x;v.push_back(x);
			}
			m.push_back(v);
		}
		int r=1000000;
		for(int i=0;i<m.size();i++){
			int s=15*m[i].size();
			for(int j=0;j<m[i].size();j++){
				s+=m[i][j]*5;
			}
			r=min(r,s);
		}
		cout<<r<<endl;
	}
	return 0;
}

File: online-judge-solutions/Codeforces/408A.py
#https://codeforces.com/problemset/problem/408/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n=int(input())
k=list(map(int,input().split()))
m=[list(map(int,input().split())) for i in range(len(k))]
r=1000000
for i in range(len(m)):
    s=15*len(m[i])
    s+=sum([m[i][j]*5 for j in range(len(m[i]))])
    r=min(r,s)
print(r)
